This pull request doesn't compile on Windows 10 because :

1) ` FileDescriptor _fd { -1 };` (Line 77: openZia/Packet.hpp)
FileDescriptor initialization to -1
2) `#include <winsock2.h>` create the follow errors 

![](https://zupimages.net/up/20/08/l0ho.png)

`D:\Windows Kits\10\include\10.0.18362.0\shared\ws2def.h(103): warning C4005: 'AF_IPX': macro redefinition`
`D:\Windows Kits\10\include\10.0.18362.0\um\winsock.h(457): note: see previous definition of 'AF_IPX'`

macro/variables from winsock2.h might be already included in another header. So it might be unnessary to include this header.

I reinforced openZia tests to see if it compiles on Windows in advance (pull request link #27)

--- a/openZia/Endpoint.hpp
+++ b/openZia/Endpoint.hpp
@@ -12,17 +12,13 @@
 #include <cinttypes>
 #include "OperatingSystem.hpp"
 
-#if defined(SYSTEM_WINDOWS)
-    #include <winsock2.h>
-#endif
-
 namespace oZ
 {
     class Endpoint;
 
     using IP = std::uint32_t;
     using Port = std::uint16_t;
-    
+
     #if defined(SYSTEM_LINUX) || defined(SYSTEM_DARWIN)
         using FileDescriptor = std::int32_t;
     #elif defined(SYSTEM_WINDOWS)

--- a/openZia/Packet.hpp
+++ b/openZia/Packet.hpp
@@ -74,6 +74,12 @@ public:
 private:
     ByteArray _byteArray {};
     Endpoint _endpoint {};
-    FileDescriptor _fd { -1 };
+
+    #if defined(SYSTEM_LINUX) || defined(SYSTEM_DARWIN)
+        FileDescriptor _fd { -1 };
+    #elif defined(SYSTEM_WINDOWS)
+        FileDescriptor _fd = INVALID_SOCKET;
+    #endif
+
     bool _useEncryption = false;
 };

